### PR TITLE
SCT-242: add assigner (User) to MissionTaskUserAssignment

### PR DESCRIPTION
**Pull request by naknomum**
## Pull Request Overview

- Adds `assigner` (User, _required_) to `MissionTaskUserAssignment` to record _who_ did the assigning
- DB update to support same
- Modifications to schema for `/api/v1/users/me` and `/api/v1/missions/tasks/GUID` to also add the `assigner` to listed MissionTasks

---

**Review Notes**
- :warning: DB migration upgrade will _fail_ if there are **existing MissionTaskUserAssignments** due to the fact that the added `assigner_guid` column is not-null.  Best solution is just to blow away any existing task assignments first, as upgrade will work fine on an empty table.

### CODE REVIEW

File: migrations/versions/2022b01697e4_.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+
+def upgrade():
+    """
+    Upgrade Semantic Description:
+        ENTER DESCRIPTION HERE
+    """
+    # ### commands auto generated by Alembic - please adjust! ###
+    with op.batch_alter_table('mission_task_user_assignment', schema=None) as batch_op:
+        batch_op.add_column(
```
Comment (hwindsor) [inline migrations/versions/2022b01697e4_.py:27]: Can't this be done as a two step operation, create it as nullable=True. 
But then I guess, what user guid do you put here? Not sure

File: app/modules/missions/models.py [hunk 2]
```diff
@@ -538,17 +548,29 @@ def user_is_assigned(self, user):
     def get_assigned_users(self):
         return [assignment.user for assignment in self.user_assignments]
 
+    def get_assigned_users_with_assigner_json(self):
```
Comment (hwindsor) [inline app/modules/missions/models.py:551]: always slightly twitchy with hand rolled json. Souldn't this be done by a schema?

Comment (naknomum) [reply app/modules/missions/models.py:551]: agreed. i couldnt think how to do it.  basically this _was_ (and still kind of is) the **schema for a list of users**.  however, for each user there is added a property called `assigner` which is _itself_ a user(!)... but this user comes from the **assignment** that designated the top-level user to be attached to this task.  i could not figure out how to get the schema to be passed this extra bit of information.   (may take up in slack with you - would love to know the magic to make this happen.)

File: migrations/versions/2022b01697e4_.py (hunk 1 above)
Comment (naknomum) [reply migrations/versions/2022b01697e4_.py:27]: exactly - this is the problem we have had before with nullable=False for _foreign keys_.  (if it was just a simple value like int or string we could do as you describe:  nullable=True, set to some "acceptable" dummy value, then nullable=False.)  but when its pointing to something like a user, not so much.

**Review by hwindsor — APPROVED** [review]
LGTM

**Pull request merged**